mono/mini/exceptions-arm64.c

```diff
@@ -620,3 +626,598 @@ mono_arch_do_ip_adjustment (MonoContext *ctx)
+	g_assert(offset <= max_offset);
+	if (unwind_codes_buffer_size + unwind_code_size > MONO_ARM64_MAX_UNWIND_CODE_SIZE)
+		g_error ("Larger allocation needed for the unwind information.");
+}
+
+static void
+mono_arch_unwind_add_nop (guint8* unwind_codes, guint32* unwind_code_size) {
+	mono_arch_unwind_add_assert(1, 0, *unwind_code_size, 0);
+	unwind_codes[(*unwind_code_size)++] = 0b11100011;
+}
+
+static void
+mono_arch_unwind_add_nops (guint8* unwind_codes, guint32* unwind_code_size, int nop_count) {
+	while (nop_count--)
+		mono_arch_unwind_add_nop (unwind_codes, unwind_code_size);
+}
+
+
+static void
+mono_arch_unwind_add_set_fp (guint8* unwind_codes, guint32* unwind_code_size) {
+	mono_arch_unwind_add_assert (1, 0, *unwind_code_size, 0);
+	unwind_codes[(*unwind_code_size)++] = 0b11100001;
+}
+
+static void
+mono_arch_unwind_add_pac_sign_lr (guint8* unwind_codes, guint32* unwind_code_size) {
+	mono_arch_unwind_add_assert (1, 0, *unwind_code_size, 0);
+	unwind_codes[(*unwind_code_size)++] = 0b11111100;
+}
+
+
+static void
+mono_arch_unwind_add_end (guint8* unwind_codes, guint32* unwind_code_size) {
+	mono_arch_unwind_add_assert (1, 0, *unwind_code_size, 0);
+	unwind_codes[(*unwind_code_size)++] = 0b11100100;
+}
+
+static void
+mono_arch_unwind_add_end_c (guint8* unwind_codes, guint32* unwind_code_size) {
+	mono_arch_unwind_add_assert (1, 0, *unwind_code_size, 0);
+	unwind_codes[(*unwind_code_size)++] = 0b11100101;
+}
+
+static void
+mono_arch_unwind_add_save_fplr (guint8* unwind_codes, guint32* unwind_code_size, gint32 offset) {
+
+	mono_arch_unwind_add_assert (1, 504, *unwind_code_size, offset);
+
+	unwind_codes[(*unwind_code_size)++] = 0b01000000 | offset / 8;
+}
+
+static void
+mono_arch_unwind_add_save_fplr_x (guint8* unwind_codes, guint32* unwind_code_size, gint32 offset) {
+
+	mono_arch_unwind_add_assert (1, 512, *unwind_code_size, offset);
+
+	unwind_codes[(*unwind_code_size)++] = 0b10000000 | (offset-8) / 8;
+}
+
+static void
+mono_arch_unwind_add_save_reg (guint8* unwind_codes, guint32* unwind_code_size, guint32 reg, guint32 offset, gboolean reverse) {
+
+	mono_arch_unwind_add_assert (2, 504, *unwind_code_size, offset);
+
+	guint8 reg_offset = reg - ARMREG_R19;
+	int order = reverse ? 1 : 0;
+
+	unwind_codes[*unwind_code_size + order] = 0b11010000 | reg_offset / 8;
+	unwind_codes[*unwind_code_size + 1 - order] = (reg_offset & 0x8) << 5 | (offset / 8);
+	*(unwind_code_size) += 2;
+}
+
+static void
+mono_arch_unwind_add_save_reg_x (guint8* unwind_codes, guint32* unwind_code_size, guint32 reg, guint32 offset, gboolean reverse) {
+
+	mono_arch_unwind_add_assert (2, 256, *unwind_code_size, offset);
+
+	guint8 reg_offset = reg - ARMREG_R19;
+	int order = reverse ? 1 : 0;
+
+	unwind_codes[*unwind_code_size + order] = 0b11010000 | reg_offset / 8;
+	unwind_codes[*unwind_code_size + 1 - order] = (reg_offset & 0x8) << 5 | (offset / 8);
+	*(unwind_code_size) += 2;
```

Comment:
This looks like a copy of `mono_arch_unwind_add_save_reg`. The only diff I notice is the max_offset passed to the assert code.  Is this correct?

Reply:
The binary code, 0b1101000 was wrong - fixed inhttps://github.com/Unity-Technologies/mono/commit/9b8e291ee6dde9c51871bf05de6ad195e1d48f01